File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
	char start = ' ';
	int prev = 0;
	int next = 0;

	for (auto q : s)
	{
		if (start == ' ')
		{
			start = q;
			prev++;
			continue;
		}
		if (start == q)
			prev++;
		else next++;

		if (prev == next)
		{
			answer++;
			prev = 0;
			next = 0;
			start = ' ';
		}
	}

	if (prev !=0) answer++;
    return answer;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cs
using System;

public class Solution {
    public int solution(string s) {
        int answer = 0;
        char chk = ' ';
        int cnt1 = 0;
        int cnt2 = 0;
        foreach(var item in s)
        {
            if(chk == ' ')
            {
                chk = item;
                cnt1++;
            }
            else
            {
                if(chk == item)
                {
                    cnt1++;
                }
                else
                {
                    cnt2++;
                }
                
                if(cnt1 == cnt2)
                {
                    answer++;
                    cnt1 = 0;
                    cnt2 = 0;
                    chk = ' ';
                }
            }
        }
        if(chk != ' ') answer++;
        return answer;
    }
}